capstone/scripts/convert_s3.py

```diff
@@ -89,123 +89,135 @@ def export_cases_to_s3(redacted: bool, reporter_id: str) -> None:
+    vars["fake_request"] = namedtuple("Request", ["query_params", "accepted_renderer"])(
+        query_params={"body_format": "text"},
+        accepted_renderer=None,
+    )
+
+    # create a metadata contents string to append case metadata content
+    metadata_contents = b""
+
+    # store the serialized case data in tempfile
+    for item in cases_search.scan():
+        with tempfile.NamedTemporaryFile() as file:
+            # pass case in to add additional data to the CaseDocument
+            case = cases_by_id[item["_source"]["id"]]
 
-    # select corresponding case search obj
-    case_search = CaseDocument.raw_search().filter("term", id=case.id)
-    # Store the serialized data
-    with tempfile.NamedTemporaryFile() as file:
-        for item in case_search.scan():
             serializer = vars["serializer"](
-                item["_source"], context={"request": vars["fake_request"]}
+                item["_source"],
+                context={
+                    "request": vars["fake_request"],
+                    "first_page_order": case.first_page_order,
+                    "last_page_order": case.last_page_order,
+                },
             )
-            file.write(json.dumps(serializer.data).encode("utf-8") + b"\n")
+
+            # add data to metadata_contents string without 'casebody'
+            metadata_data = serializer.data
+            metadata_data.pop("casebody", None)
+            metadata_contents += json.dumps(metadata_data).encode("utf-8") + b"\n"
+
+            # compose each case file with a hash
+            case_contents = json.dumps(serializer.data).encode("utf-8") + b"\n"
+            hash_object = hashlib.sha256(case_contents)
+            case_contents_hash = base64.b64encode(hash_object.digest()).decode()
+
+            # calculate case file name
+            if prev_case_first_page == case.first_page:
+                case_file_name_index += 1
+            else:
+                case_file_name_index = 1
+            case_file_name = (
+                f"{case.first_page.zfill(4)}-{str(case_file_name_index).zfill(2)}.json"
+            )
+
+            # set so we can use to determine multiple cases on single page
+            prev_case_first_page = case.first_page
+
+            # ignore cases that have already been uploaded
+            if (
+                case_contents_hash in s3_contents_hash
+                and s3_contents_hash[case_contents_hash] == case_file_name
+            ):
+                s3_contents_hash.pop(case_contents_hash, None)
+                print(f"Skipped {cases_key}{case_file_name}")
+                continue
+            else:
+                file.write(case_contents)
+                file.flush()
+                hash_and_upload(
+                    file,
+                    dest_bucket,
+                    f"{cases_key}{case_file_name}",
+                    "application/jsonl",
+                )
+
+    with tempfile.NamedTemporaryFile() as file:
```

Comment:
None of these tempfiles are helping you, since `hash_and_upload` just reads the file back into a string. I would update `hash_and_upload` to take a `str` and remove these.